### PR TITLE
Flatbuffers - Add support for rejected CVEs

### PR DESCRIPTION
**Pull request by sebasfalcone**
|Related issue|
|---|
| #19723 |

## Description
This PR updates the CVEv5 flatbuffer schema in order to allow the serialization of rejected CVEs

Change log:
- Added rejected CVEs fields to the schema
- Updated CVSS table name
- Removed required flag from schema (prevent failure when documents are filtered)

image

> **Important**
Due to these changes, the flatbuffer is not 100% CVEv5 compliant so some weird scenarios are possible. For example, having an ADP entry on a rejected CVE:

<details>
  <summary>Show more</summary>
  
```json
{
    "containers": {
        "adp": [
            {
                "affected": [
                    {
                        "defaultStatus": "unknown",
                        "platforms": [
                            "upstream"
                        ],
                        "product": "cvs",
                        "vendor": "canonical"
                    }
                ],
                "descriptions": [
                    {
                        "lang": "en",
                        "value": "Alen Zukich discovered a buffer overflow in the processing of version and author information in the CVS client. By tricking an user to connect to a malicious CVS server, an attacker could exploit this to execute arbitrary code with the privileges of the connecting user."
                    }
                ],
                "providerMetadata": {
                    "orgId": "cc1ad9ee-3454-478d-9317-d3e869d708bc",
                    "shortName": "canonical"
                },
                "references": [
                    {
                        "url": "https://ubuntu.com/security/notices/USN-117-1"
                    }
                ]
            }
        ],
        "cna": {
            "providerMetadata": {
                "dateUpdated": "2018-03-16T14:29:07Z",
                "orgId": "00000000-0000-4000-A000-000000000003",
                "shortName": "nvd"
            },
            "rejectedReasons": [
                {
                    "lang": "en",
                    "value": "** REJECT **  DO NOT USE THIS CANDIDATE NUMBER.  ConsultIDs: none.  Reason: This candidate was in a CNA pool that was not assigned to any issues during 2017.  Notes: none."
                }
            ]
        }
    },
    "cveMetadata": {
        "assignerOrgId": "00000000-0000-4000-A000-000000000003",
        "assignerShortName": "nvd",
        "cveId": "CVE-2017-4085",
        "datePublished": "2018-03-16T14:29:07Z",
        "dateUpdated": "2018-03-16T14:29:07Z",
        "state": "REJECTED"
    },
    "dataType": "CVE_RECORD",
    "dataVersion": "5.0"
}
```
- CVEv5 schema validation:
image

- Flatbuffer conversion:
```
Parsing CVE-2017-4085.json
Success: 1
failed:  0
Total:   1
```

</details>

## Results
I've converted all the generated documents, and all the rejected ones were successfully parsed:

```
Success: 227436
failed:  0
Total:   227436
```

### CODE REVIEW

**Review by pereyra-m — APPROVED** [review]
LGTM!

I'd just add a test downloading the full content from cti

File: src/wazuh_modules/vulnerability_scanner/tests/unit/cve5FB_test.cpp [hunk 1]
```diff
@@ -0,0 +1,241 @@
+/*
+ * Wazuh SyscollectorFlatbuffers
```
Comment (tsarquis88) [inline src/wazuh_modules/vulnerability_scanner/tests/unit/cve5FB_test.cpp:2]: Shouldn't this be ... ?
```suggestion
 * Wazuh Vulnerability Scanner - Unit Tests
```

File: src/wazuh_modules/vulnerability_scanner/tests/unit/cve5FB_test.hpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Wazuh SyscollectorFlatbuffers
```
Comment (tsarquis88) [inline src/wazuh_modules/vulnerability_scanner/tests/unit/cve5FB_test.hpp:2]: Same as in the .cpp
```suggestion
 * Wazuh Vulnerability Scanner - Unit Tests
```

**Review by tsarquis88 — CHANGES_REQUESTED** [review]

**Review by tsarquis88 — APPROVED** [review]
LGTM! GJ adding some light to these schemas.

**Review by Damian-Mangold — APPROVED** [review]
LGTM !

**Pull request merged**